lib/zendesk_api/middleware/response/parse_iso_dates.rb

```diff
@@ -19,7 +19,9 @@ def parse_dates!(value)
           case value
           when Hash then value.each { |key,element| value[key] = parse_dates!(element) }
           when Array then value.each_with_index { |element, index| value[index] = parse_dates!(element) }
-          when /\A\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z\Z/m then Time.parse(value)
+          when String
+            return Time.parse(value) if value.valid_encoding? && value =~ /\A\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z\Z/m
+            value
```

Comment:
I don't see a usecase for having/keeping bad characters in a date field ...
if it were a text ... maybe ...
Any way we can fix the original issue of 'something responds with bad
characters' ?

On Fri, Sep 4, 2015 at 8:37 AM, Fábio Neves <notifications@github.com>
wrote:

> In lib/zendesk_api/middleware/response/parse_iso_dates.rb
> <https://github.com/zendesk/zendesk_api_client_rb/pull/266#discussion_r38764006>
> :
>
> > @@ -19,7 +19,9 @@ def parse_dates!(value)
> >            case value
> >            when Hash then value.each { |key,element| value[key] = parse_dates!(element) }
> >            when Array then value.each_with_index { |element, index| value[index] = parse_dates!(element) }
> > -          when /\A\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z\Z/m then Time.parse(value)
> > +          when String
> > +            return Time.parse(value) if value.valid_encoding? && value =~ /\A\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z\Z/m
> > +            value
>
> We actually sanitise the data in our project after we get the response. We
> were not sure however if we should actually do it at the this level or just
> let the clients handle it. We thought that there might be the possibility
> that there are characters the clients will be able to handle. If we do that
> we might be destroying their data.
> So, in the case this is always a case of bad data I'm up to replace this
> with a sanitise method, but as we weren't sure we just fixed the gem to not
> blow up when we get to this case.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendesk/zendesk_api_client_rb/pull/266/files#r38764006>
> .
>